apollo-ios/Sources/ApolloAPI/GraphQLOperation.swift

```diff
@@ -155,6 +143,25 @@ public extension GraphQLSubscription {
   @inlinable static var operationType: GraphQLOperationType { return .subscription }
 }
 
+// MARK: - OperationResponseFormat
+
+/// The format expected for the network response of an operation.
+public protocol OperationResponseFormat {}
+
+/// A response format for an operation that expects a single network response.
+///
+/// This is the default format for most operations.
+public struct SingleResponseFormat: OperationResponseFormat {}
+
+/// An incremental response format for an operation which uses the `@defer` directive.
+public struct IncrementalDeferredResponseFormat: OperationResponseFormat {
+  public let deferredFragments: [DeferredFragmentIdentifier: any SelectionSet.Type]
+
+  public init(deferredFragments: [DeferredFragmentIdentifier: any SelectionSet.Type]) {
+    self.deferredFragments = deferredFragments
+  }
+}
```

Comment:
I think we're missing a response format for HTTP subscriptions? They're closely related to `IncrementalDeferredResponseFormat` but not defined by the use of `@defer`.

Reply:
You're right! I'll be getting to that soon in another PR.